util/Pos/Util/Util.hs

```diff
@@ -388,3 +390,26 @@ median l = NE.sort l NE.!! middle
 -}
 sleep :: MonadIO m => NominalDiffTime -> m ()
 sleep n = liftIO (threadDelay (truncate (n * 10^(6::Int))))
+
+-- | 'tMeasure' with 'logDebug'.
+tMeasureLog :: (MonadIO m, WithLogger m) => Text -> m a -> m a
+tMeasureLog = tMeasure logDebug
+
+-- | 'tMeasure' with 'putText'. For places you don't have
+-- 'WithLogger' constraint.
+tMeasureIO :: (MonadIO m) => Text -> m a -> m a
+tMeasureIO = tMeasure putText
+
+-- | Measures time executed by the particular action, logs it before
+-- returning the result.
```

Comment:
Should add: also forces the result before taking the second time sample, and before logging.

Reply:
Fixed.